python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -1121,10 +1121,15 @@ def getJobObject(self):
         Unknown: Why is this a GangaObject method and not a Node method?
         """
         from Ganga.GPIDev.Lib.Job import Job
-        r = self._getRoot(cond=lambda o: isinstance(o, Job))
-        if not isinstance(r, Job):
-            raise AssertionError('no job associated with object ' + repr(self))
-        return r
+        if self._getParent():
+            r = self._getRoot(cond=lambda o: isinstance(o, Job))
+            if not isinstance(r, Job):
+                raise AssertionError('No Job associated with object instead root=\'%s\' for \'%s\'' % (repr(self), type(r)))
```

Comment:
Are the arguments the wrong way round here? `root` is set to `repr(self)` rather than `type(r)` (`r` being the return of `_getRoot`).